File: app/src/main/java/com/yawar/memo/modelView/VerficationViewModel.java
package com.yawar.memo.modelView;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.yawar.memo.model.UserModel;
import com.yawar.memo.repositry.AuthRepo;
import com.yawar.memo.utils.BaseApp;
import org.json.JSONObject;

public class VerficationViewModel extends ViewModel {
    BaseApp baseApp = BaseApp.getInstance();
    AuthRepo authRepo = baseApp.getAuthRepo();

    public MutableLiveData<JSONObject> getSpecialNumber(){
        return authRepo.jsonObjectMutableLiveData;
    }
    public MutableLiveData<Boolean> getLoading(){
        return authRepo.loading;
    }

    public MutableLiveData<Boolean> getErrorMessage(){
        return authRepo.showErrorMessage;
    }
    public void setLoading(Boolean check){
        authRepo.loading.setValue(check);
    }
    public void setErrorMessage(Boolean check){
        authRepo.showErrorMessage.setValue(check);
    }









}

File: app/src/main/java/com/yawar/memo/modelView/VerficationViewModel.kt
package com.yawar.memo.modelView

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.yawar.memo.utils.BaseApp
import org.json.JSONObject

class VerficationViewModel : ViewModel() {
    var baseApp = BaseApp.getInstance()
    var authRepo = baseApp.authRepo

    fun getSpecialNumber(): LiveData<JSONObject?> {
        return authRepo.jsonObjectMutableLiveData
    }

        fun getLoading(): LiveData<Boolean> {
            return authRepo.loadingMutableLiveData
        }

        fun getErrorMessage(): LiveData<Boolean> {
            return authRepo.showErrorMessage
        }
    fun setLoading(check: Boolean) {
        authRepo.setLoading(check)
    }
    fun setErrorMessage(check: Boolean) {
        authRepo.setShowErrMessage(check)
    }


}